pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,184 @@
+installation on Windows, but not other platforms.
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to give some information about
+the conditions under which a requirement should be installed.
+
+
+Examples
+========
+
+Here are some examples of such markers::
+
+   "sys_platform == 'win32'"
+   "platform_machine == 'i386'"
+   "python_version == '2.4' or python_version == '2.5'"
+   "python_version>='2.7' and python_version<'3.0'"
+   "'linux' in sys_platform"
+
+And here's an example of some conditional metadata for a distribution that
+requires PyWin32 both at runtime and buildtime when using Windows::
+
+    "name": "ComfyChair",
+    "run_requires": [
+      {
+        "requires": ["pywin32 > 1.0"],
+        "environment": "sys.platform == 'win32'"
+      }
+    ]
+    "build_requires": [
+      {
+        "requires": ["pywin32 > 1.0"],
+        "environment": "sys.platform == 'win32'"
+      }
+    ]
+
+
+Micro-language
+==============
+
+The micro-language behind this is a simple subset of Python. It compares::
+
+* strings with the ``==`` and ``in`` operators (and their opposites)
+* version numbers with the ``<``, ``<=``, ``>=``, and ``<`` operators
+  in addition to those supported for strings
+
+The usual boolen operators ``and`` and ``or`` can be used to combine
+expressions, and parentheses are supported for grouping.
+
+The pseudo-grammar is not ::
+
+    MARKER: EXPR [(and|or) EXPR]*
+    EXPR: ("(" MARKER ")") | (SUBEXPR [CMPOP SUBEXPR])
+    CMPOP: (==|!=|<|>|<=|>=|in|not in)
+
+but my memory of BNF isn't quite up to fixing it just now.
+
+``SUBEXPR`` is either a Python string (such as ``'2.4'``, or
+``'win32'``) or one of the below marker variables.
+
+Version numbers
+---------------
+
+* ``python_version``: ``'{0.major}.{0.minor}'.format(sys.version_info)``
+* ``python_full_version``: see definition below
+* ``platform_version``: ``platform.version()``
+* ``implementation_version````: see definition below
+
+All version numbers must be valid PEP-0440[3] identifiers, and will be
+compared using PEP-0440 semantics.
+
+Strings
+-------
+
+* ``os_name````: ``os.name``
+* ``sys_platform````: ``sys.platform``
+* ``platform_release``: ``platform.release()``
+* ``implementation_name````: ``sys.implementation.name``
+* ``platform_machine``: ``platform.machine()``
+* ``platform_python_implementation``: ``platform.python_implementation()``
+
+
+If a particular value is not available (such as the ``sys.implementation``
+subattributes in versions of Python prior to 3.3), the corresponding marker
+variable MUST be considered equivalent to the empty string.
+
+Note that it is not possible to use other sequences like tuples or
+lists on the right side of the ``in`` and ``not in`` operators,
+regardless of whether the comparison is with a version number, string,
+or one of the marker variable names (which either refers to a string
+or a variable number)
```

Comment:
I think this can go if we remove the 'subset of Python' framing higher up. It becomes clear, like 'argparse>=2.3,<3' is not valid Python, that neither is this.